irohad/main/impl/storage_init.cpp

```diff
@@ -76,6 +78,44 @@ namespace {
   }
 }  // namespace
 
+iroha::expected::Result<std::shared_ptr<iroha::ametsuchi::Storage>, std::string>
+iroha::initStorage(
+    std::shared_ptr<ametsuchi::RocksDBContext> db_context,
+    std::shared_ptr<iroha::PendingTransactionStorage> pending_txs_storage,
+    std::shared_ptr<shared_model::interface::QueryResponseFactory>
+        query_response_factory,
+    boost::optional<std::string> block_storage_dir,
```

Comment:
can we stick to std optional? it looks a bit strange when we have boost::optional and std::optional on the next line)

Reply:
It seems to be the purpose for future refactoring.